src/util/nextThought.js

```diff
@@ -0,0 +1,240 @@
+  nextSibling as thoughtNextSibling,
+  unroot,
+} from '../util.js'
+
+/**
+ * Returns true if the context view is activated and has more than one context
+ * @returns {boolean}
+ */
+const isValidContextView = path =>
+  path.length &&
+  isContextViewActive(pathToContext(path)) &&
+  getContexts(head(path).value).length > 1
+
+/**
+ * Adds the rank of the child thought to every thought in a context.
+ * @param {ContextInfo} contextInfo.   A { context, rank } object returned from getContexts.
+ * @returns {array | null} rankedContext
+ */
+const contextWithThoughtRank = contextInfo => {
+  return contextInfo && contextInfo.context.map(value => {
+    const thought = getThought(value)
+    return { ...(thought || { value }), rank: contextInfo.rank }
+  })
+}
+
+/**
+ * Returns the next sibling of the focused context within a context view.
+ * @returns {array | null} rankedContext
+ */
+const nextSiblingContext = (rank, context, thoughtIndex) => {
+  const contextSiblings = getContextsSortedAndRanked(head(context))
+  const currentContextIndex = contextSiblings.findIndex(context => context.rank === rank)
+  // const currentIndex = contextSiblings.findIndex(thoughts => thoughts.find(thought => thought.value === value))
+  const nextSibling = contextSiblings[currentContextIndex + 1]
+    ? contextSiblings[currentContextIndex + 1]
+    : null
+  return contextWithThoughtRank(nextSibling, thoughtIndex)
+}
+
+/**
+ * Returns the first child of context at the given path
+ * @returns {array | undefined} rankedContext
+ */
+const firstChildOfContextView = (path, thoughtIndex) => {
+  const context = pathToContext(path)
+  const contextChildren = getContextsSortedAndRanked(head(context))
+  const firstChild = contextChildren[0]
+  return contextWithThoughtRank(firstChild, thoughtIndex)
+}
+
+/**
+ * Returns the first visible child of thought at the given path
+ * @returns {object | undefined} thought
+ */
+const firstChildOfThoughtView = (path, showHiddenThoughts) => {
+  const contextMeta = meta(path)
+  const sortPreference = getSortPreference(contextMeta)
+  const children = (sortPreference === 'Alphabetical' ? getThoughtsSorted : getThoughtsRanked)(path)
+  const notHidden = child => !isFunction(child.value) && !meta(pathToContext(path).concat(child.value)).hidden
+  return showHiddenThoughts ? children[0] : children.find(notHidden)
+}
+
+/**
+ * Returns the path to the current thought by stripping out any context views
+ * @returns {array} path
+ */
+const getPathFromContextChain = contextChain => {
+  // last of second last item in context chain gives us the current context
+  const context = head(head(contextOf(contextChain)))
+  const contexts = getContextsSortedAndRanked(context.value)
+  const currentContextTop = head(contextChain)[0].value
+  const matchedContextChain = contexts.find(c => c.context.includes(currentContextTop))
+  const matchedContext = matchedContextChain.context.map(i => ({ value: i, rank: matchedContextChain.rank }))
```

Comment:
Setting `rank` here causes the same issue described above. It should only be applied to the relevant thought, not all thoughts.